presto-main/src/main/java/com/facebook/presto/server/StatementResource.java

```diff
@@ -210,6 +216,23 @@ private static Response getQueryResults(Query query, Optional<Long> token, UriIn
         query.getResetSessionProperties().stream()
                 .forEach(name -> response.header(PRESTO_CLEAR_SESSION, name));
 
+        // add added prepare statements
+        query.getAddedPreparedStatements().entrySet().stream()
+                .forEach(entry -> {
+                    try {
+                        String encodedKey = encode(entry.getKey(), "UTF-8");
+                        String encodedValue = encode(entry.getValue(), "UTF-8");
+                        response.header(PRESTO_ADDED_PREPARE, encodedKey + '=' + encodedValue);
+                    }
+                    catch (UnsupportedEncodingException e) {
+                        throw new PrestoException(UNSUPPORTED_ENCODING, "Cannot encode statement: UTF-8 encoding unsupported");
```

Comment:
This shouldn't be a PrestoException. If "UTF-8" is unsupported, something is really wrong with the underlying environment/VM. Throw an AssertionError instead.